src/Breadcrumb/CreateAppForDeveloperBreadcrumbBuilder.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Breadcrumb;
+
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\user\Entity\User;
+
+/**
+ * Provides a breadcrumb builder for create developer app for developer page.
+ */
+class CreateAppForDeveloperBreadcrumbBuilder implements BreadcrumbBuilderInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    return $route_match->getRouteName() === 'entity.developer_app.add_form_for_developer';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(RouteMatchInterface $route_match) {
+    $breadcrumb = new Breadcrumb();
+    $user = $route_match->getParameter('user');
+
+    $breadcrumb->addLink(Link::createFromRoute($this->t('Home'), '<front>'));
+    $breadcrumb->addLink(Link::createFromRoute(User::load($user)->getUsername(), 'entity.user.canonical', [
+      'user' => $user,
+    ]));
+    $breadcrumb->addLink(Link::createFromRoute(
+      t('My @developer_app', [
```

Comment:
Couldn't you leverage the `_title_callback` of the `entity.developer_app.collection_by_developer` route somehow here? This seems a code duplication to me and if the default page title changes in `\Drupal\apigee_edge\Controller\DeveloperAppListBuilderForDeveloper::getPageTitle` or the `_title_callback` of the route gets altered then the title of the page and what a user can see in the breadcrumb is not going to be aligned.